tests/components/plex/test_browse_media.py

```diff
@@ -0,0 +1,165 @@
+
+    tvshows = next(iter(x for x in result["children"] if x["title"] == "TV Shows"))
+    playlists = next(iter(x for x in result["children"] if x["title"] == "Playlists"))
+
+    # Browse into a Plex TV show library
+    msg_id += 1
+    await websocket_client.send_json(
+        {
+            "id": msg_id,
+            "type": "media_player/browse_media",
+            "entity_id": media_players[0],
+            ATTR_MEDIA_CONTENT_TYPE: tvshows[ATTR_MEDIA_CONTENT_TYPE],
+            ATTR_MEDIA_CONTENT_ID: str(tvshows[ATTR_MEDIA_CONTENT_ID]),
+        }
+    )
+
+    msg = await websocket_client.receive_json()
+    assert msg["id"] == msg_id
+    assert msg["type"] == TYPE_RESULT
+    assert msg["success"]
+    result = msg["result"]
+    assert result[ATTR_MEDIA_CONTENT_TYPE] == "library"
+    result_id = result[ATTR_MEDIA_CONTENT_ID]
+    assert len(result["children"]) == len(
+        mock_plex_server.library.sectionByID(result_id).all()
+    )
+
+    # Browse into a Plex TV show
+    msg_id += 1
+    await websocket_client.send_json(
+        {
+            "id": msg_id,
+            "type": "media_player/browse_media",
+            "entity_id": media_players[0],
+            ATTR_MEDIA_CONTENT_TYPE: result["children"][0][ATTR_MEDIA_CONTENT_TYPE],
+            ATTR_MEDIA_CONTENT_ID: str(result["children"][0][ATTR_MEDIA_CONTENT_ID]),
+        }
+    )
+
+    msg = await websocket_client.receive_json()
+    assert msg["id"] == msg_id
+    assert msg["type"] == TYPE_RESULT
+    assert msg["success"]
+    result = msg["result"]
+    assert result[ATTR_MEDIA_CONTENT_TYPE] == "show"
+    result_id = int(result[ATTR_MEDIA_CONTENT_ID])
+    assert result["title"] == mock_plex_server.fetchItem(result_id).title
+
+    # Browse into a non-existent TV season
+    msg_id += 1
+    await websocket_client.send_json(
+        {
+            "id": msg_id,
+            "type": "media_player/browse_media",
+            "entity_id": media_players[0],
+            ATTR_MEDIA_CONTENT_TYPE: result["children"][0][ATTR_MEDIA_CONTENT_TYPE],
+            ATTR_MEDIA_CONTENT_ID: str(99999999999999),
+        }
+    )
+
+    msg = await websocket_client.receive_json()
+    assert msg["id"] == msg_id
+    assert msg["type"] == TYPE_RESULT
+    assert not msg["success"]
+    assert msg["error"]["code"] == ERR_UNKNOWN_ERROR
+
+    # Browse Plex playlists
+    msg_id += 1
+    await websocket_client.send_json(
+        {
+            "id": msg_id,
+            "type": "media_player/browse_media",
+            "entity_id": media_players[0],
+            ATTR_MEDIA_CONTENT_TYPE: playlists[ATTR_MEDIA_CONTENT_TYPE],
+            ATTR_MEDIA_CONTENT_ID: str(playlists[ATTR_MEDIA_CONTENT_ID]),
+        }
+    )
+
+    msg = await websocket_client.receive_json()
+    assert msg["id"] == msg_id
+    assert msg["type"] == TYPE_RESULT
+    assert msg["success"]
+    result = msg["result"]
+    assert result[ATTR_MEDIA_CONTENT_TYPE] == "playlists"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6072450)